packages/messaging/src/controllers/window-controller.ts

```diff
@@ -133,13 +136,15 @@ export class WindowController implements FirebaseMessaging, FirebaseService {
    */
   // TODO: Simplify this to only accept a function and not an Observer.
   onMessage(nextOrObserver: NextFn<object> | Observer<object>): Unsubscribe {
-    this.onMessageCallback =
-      typeof nextOrObserver === 'function'
-        ? nextOrObserver
-        : nextOrObserver.next;
+    if (typeof nextOrObserver === 'function') {
+      this.onMessageCallback = nextOrObserver;
+    } else {
+      this.onMessageObserver = nextOrObserver;
+    }
```

Comment:
This allows registering 2 listeners and is possibly a breaking change. I don't think we should change the current behavior of single listener. On way to do it can be remove `onMessageCallback` field, convert the callback to an observer and assign it to `onMessageObserver`.

Reply:
I can't think of a case where this could break an application. Maybe some unexpected behaviors? But I agree that this can be done more elegantly with one callback since there is really no need to have two.